### PR TITLE
Add config to pass in environment variables

### LINKED ISSUE #205: Proxy support in called commands
Unfortunately I have to live behind a corporate proxy, which means that the commands that have to traverse that proxy need to be aware of it.

Normall just setting HTTP_PROXY/http_proxy should be enough in the environment, but please doesn't pass it through (https://github.com/thought-machine/please/blob/master/src/core/build_env.go)

Pip supports a `--proxy` but some other tools don't.

### PR DESCRIPTION
**Pull request by jtdoepke**
This adds a config `Build.PassEnv` to copy environment variables from the user's current shell. Example:

```
[Build]
PassEnv = http_proxy
PassEnv = HTTP_PROXY
```

Should fix #205 in most cases, I think.

### CODE REVIEW

File: src/core/config.go [hunk 1]
```diff
@@ -435,6 +437,19 @@ func (config *Configuration) GetBuildEnv() []string {
 	return config.buildEnvStored
 }
 
+// GetPassEnv returns environment variables from the user's shell.
+func (config *Configuration) GetPassEnv() []string {
```
Comment (peterebden) [inline src/core/config.go]: Would be nice to add this to GetBuildEnv as well so it only has to build the slice once.

They should also form part of the config hash which that will take care of.

Comment (jtdoepke) [reply src/core/config.go]: Just combined into the same function, something like this?
```go
// GetBuildEnv returns the build environment configured for this config object.
func (config *Configuration) GetBuildEnv() []string {
	config.buildEnvOnce.Do(func() {
		config.buildEnvStored = []string{}

		// from the BuildEnv config keyword
		for k, v := range config.BuildEnv {
			pair := strings.Replace(strings.ToUpper(k), "-", "_", -1) + "=" + v
			config.buildEnvStored = append(config.buildEnvStored, pair)
		}

		// from the user's environment based on the PassEnv config keyword
		for _, k := range config.Build.PassEnv {
			v, isSet := os.LookupEnv(k)
			if isSet {
				config.buildEnvStored = append(config.buildEnvStored, k+"="+v)
			}
		}

		sort.Strings(config.buildEnvStored)
	})
	return config.buildEnvStored
}
```

Comment (peterebden) [reply src/core/config.go]: Yup, exactly!

**Comment by peterebden** [comment]
Looks great, thanks!

**Pull request merged**